zone.go

```diff
@@ -0,0 +1,244 @@
+package dns
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"sort"
+	"strings"
+)
+
+// AddDomain adds origin to s if s is not already a FQDN.
+func AddDomain(origin, s string) string {
```

Comment:
[ Quoting <notifications@github.com> in "Re: [dns] Added new functions: Trim..." ]
>Logically one can't know if utility functions like IsEdns0, AddDomain, TrimDomain are used by project outside of this module.  I agree that documenting them would help people find them.  I know that getting started using the module was difficult for me because the documentation was so brief.
>
>Also, while a function such as IsEdns0() may seem obvious to some, the minutiae of DNS is staggeringly huge. Such utility functions prevent stupid bugs from haunting the internet.  For example without CountLabel, most people would just use strings.Count(s, "."), without IsSubDomain most people would just use strings.IsSubDomain(s, domain).
>
>That said...
>
>It sounds to me like this project only wants contributions in a very specific domain of functionality.  I'm not sure that that domain is, but I think it would make the project easier to contribute to if that domain was described in the README.md.  Feel free to contact me off-line if you'd like to know what I think it is, based on reverse engineering the comments posted on this PR.

Fair enough. Having a more offline discussion makes sense. And I of course
welcome contributions, but I also don't want to bloat the API. This goes back
to the line (that is in the README :) ):

  If there is stuff you should know as a DNS programmer there isn't a convenience
  function for it.

But that is of course really open to all kinds of interpretation(s).

I don't want to alienate people by not accepting PRs and I don't want to make Go
DNS a "everything and the kitchen sink" kind of package. So it's a balance.

Happy to discuss this some more.

/Miek

--
Miek Gieben